lib/turn.rb

```diff
@@ -54,34 +54,59 @@ def row_win
+    arr5 = [board.columns[3].rows[5].player, board.columns[4].rows[4].player, board.columns[5].rows[3].player, board.columns[6].rows[2].player]
+    right_up = [arr0, arr1, arr2, arr3, arr4, arr5]
+    
+    #key checks are in arr0[0], arr1[1], arr2[2], arr3[2], arr4[1], arr5[0]
+    #corresponding: right_up[0][0], right_up[1][1], right_up[2][2], right_up[3][2], right_up[4][1], right_up[5][0]
+    check_right = [right_up[0][0], right_up[1][1], right_up[2][2], right_up[3][2], right_up[4][1], right_up[5][0]]
+    
+  
+   if check_right.any? { |occupied| occupied.size > 0 }
+      right_up.each do |arr|
+        if arr.chunk_while{ |a, b| a == b && a != "" }.any?{ |player_piece| player_piece.size == 4}       
+          return arr.max_by{ |player| arr.count(player)}
+        end             
+      end    
+    end
+    :no_win
   end
 
   def diagonal_leftup
     #these represent the six arrays that a left_up diagonal win can occur in
-    arr1 = [board.columns[3].rows[5], board.columns[2].rows[4], board.columns[1].rows[3], board.columns[0].rows[2]]
-    arr2 = [board.columns[4].rows[5], board.columns[3].rows[4], board.columns[2].rows[3], board.columns[1].rows[2], board.columns[0].rows[1]]
-    arr3 = [board.columns[5].rows[5], board.columns[4].rows[4], board.columns[3].rows[3], board.columns[2].rows[2], board.columns[1].rows[1], board.columns[0].rows[0]]
-    arr4 = [board.columns[6].rows[5], board.columns[5].rows[4], board.columns[4].rows[3], board.columns[3].rows[2], board.columns[2].rows[1], board.columns[1].rows[0]]
-    arr5 = [board.columns[6].rows[4], board.columns[5].rows[3], board.columns[4].rows[2], board.columns[3].rows[1], board.columns[2].rows[0]]
-    arr6 = [board.columns[6].rows[3], board.columns[5].rows[2], board.columns[4].rows[1], board.columns[3].rows[0]]
-
-    left_up = [arr1, arr2, arr3, arr4, arr5, arr6]  
-
+    arr0 = [board.columns[3].rows[5].player, board.columns[2].rows[4].player, board.columns[1].rows[3].player, board.columns[0].rows[2].player]
+    arr1 = [board.columns[4].rows[5].player, board.columns[3].rows[4].player, board.columns[2].rows[3].player, board.columns[1].rows[2].player, board.columns[0].rows[1].player]
+    arr2 = [board.columns[5].rows[5].player, board.columns[4].rows[4].player, board.columns[3].rows[3].player, board.columns[2].rows[2].player, board.columns[1].rows[1].player, board.columns[0].rows[0].player]
+    arr3 = [board.columns[6].rows[5].player, board.columns[5].rows[4].player, board.columns[4].rows[3].player, board.columns[3].rows[2].player, board.columns[2].rows[1].player, board.columns[1].rows[0].player]
+    arr4 = [board.columns[6].rows[4].player, board.columns[5].rows[3].player, board.columns[4].rows[2].player, board.columns[3].rows[1].player, board.columns[2].rows[0].player]
+    arr5 = [board.columns[6].rows[3].player, board.columns[5].rows[2].player, board.columns[4].rows[1].player, board.columns[3].rows[0].player]
+    left_up = [arr0, arr1, arr2, arr3, arr4, arr5]
```

Comment:
For assessor or viewer: this is where we check the diagonal going top left down to bottom right of the board.